digest/verifiers.go

```diff
@@ -21,26 +22,26 @@ type Verifier interface {
 	// the digest.
 	Verified() bool
 
-	// Planned methods:
+	// TODO: Planned methods:
 	// Err() error
 	// Reset()
 }
 
 // NewDigestVerifier returns a verifier that compares the written bytes
 // against a passed in digest.
-func NewDigestVerifier(d Digest) Verifier {
+func NewDigestVerifier(d Digest) (Verifier, error) {
```

Comment:
Validating based on the regexp? If the algorithm is not support, this will return error.  If the digest is invalid, the verification will fail.